src/tests/test_frame_window_manager.cpp

```diff
@@ -27,41 +32,41 @@ namespace geom = mir::geometry;
 namespace
 {
 auto constexpr DISPLAY_RECT = geom::Rectangle({0, 0}, {800, 600});
-
-class MockLayoutDataAccessor : public LayoutDataAccessor
-{
-public:
-    MOCK_METHOD(std::shared_ptr<LayoutMetadata>, layout_metadata, (), (override));
-};
 }
 
 class FrameWindowManagerTest : public mtf::WindowManagementTestHarness
 {
 public:
-    FrameWindowManagerTest()
+    FrameWindowManagerTest() : runner(argc, argv)
     {
-        ON_CALL(*mock_layout_data_accessor, layout_metadata()).WillByDefault(testing::Return(nullptr));
     }
+
     auto get_builder() -> mir_test_framework::WindowManagementPolicyBuilder override
     {
         return [&](miral::WindowManagerTools const& tools)
         {
             return std::make_unique<FrameWindowManagerPolicy>(
                 tools,
                 observer,
-                mock_layout_data_accessor);
+                get_display_config());
         };
     }
 
     auto get_initial_output_configs() -> std::vector<mir::graphics::DisplayConfigurationOutput> override
     {
-        auto r = output_configs_from_output_rectangles({DISPLAY_RECT});
-        return r;
+        return output_configs_from_output_rectangles({DISPLAY_RECT});
+    }
+
+protected:
+    virtual miral::DisplayConfiguration get_display_config()
+    {
+        return miral::DisplayConfiguration(runner);
     }
 
     WindowManagerObserver observer;
-    std::shared_ptr<MockLayoutDataAccessor> mock_layout_data_accessor
-        = std::make_shared<NiceMock<MockLayoutDataAccessor>>();
+    int const argc = 1;
+    const char* argv[1] = {"test"};
```

Comment:
I get this probably works, but for main() argv is guaranteed to be nul terminated
```suggestion
    const char* argv[] = {"test", nullptr};
```